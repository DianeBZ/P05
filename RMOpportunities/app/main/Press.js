import React, { Component } from 'react';
import {
   StyleSheet,
   Text,
   Image,
   View,
   Dimensions,
   ScrollView,
   TouchableHighlight,
   BackAndroid,
} from 'react-native';

import LinearGradient from 'react-native-linear-gradient';

var width= Dimensions.get('window').width;
var height = Dimensions.get('window').height;

import Translation from './Translation';

export default class Sell extends Component {
   constructor(props) {
		super(props);
	}
   render() {
       BackAndroid.addEventListener('Back',this.onBackAndroid);
      return (
	  <ScrollView>
         <View style={styles.container}>
            <View style={styles.header} >
               <LinearGradient
                  start={{x: 0.0, y: 1.0}} end={{x: 1.0, y: 1.0}}
                  locations={[0,1.0]}
                  colors={['#185177', '#F6D092']}
                  style={styles.gradient}>
                  <View style={styles.headerIcon}>
                     <Image source={require('../img/News.png')} style = {styles.logoLeft} />
                  </View>
                  <Text style={styles.title}>
                     {Translation[lang].press_title}
                  </Text>
                  <View style={styles.headerIcon}>
                     <Image source={require('../img/TV.png')} style = {styles.logoRight} />
                  </View>
               </LinearGradient>
			</View>
			<View style={styles.containerV}>
				<View style={styles.element1}>
					<Image source={require('../img/info-chimie.png')} style = {styles.pictureI} />
				</View>
				<View style={styles.element2}>
					<Text style={styles.name}>Info chimie magazine </Text>
					<Text style={styles.teamText}>{Translation[lang].info_chimie}</Text>
				</View>
			</View>
			<View style={styles.containerV}>
				<View style={styles.element1}>
					<Image source={require('../img/eclaira.png')} style = {styles.pictureE} />
				</View>
				<View style={styles.element2}>
					<Text style={styles.name}>Eclaira  </Text>
					<Text style={styles.teamText}>{Translation[lang].eclaira}</Text>
				</View>
			</View>
			<View style={styles.containerV}>
				<View style={styles.element1}>
					<Image source={require('../img/cci-paris.png')} style = {styles.pictureC} />
				</View>
				<View style={styles.element2}>
					<Text style={styles.name}>CCI Paris Ile de France </Text>
					<Text style={styles.teamText}>{Translation[lang].cci_paris}</Text>
				</View>
			</View>
			<View style={styles.containerV}>
				<View style={styles.elementT}>
					<Image source={require('../img/telecom.png')} style = {styles.pictureT} />
				</View>
				<View style={styles.element2}>
					<Text style={styles.name}>Télécom Saint-Etienne  </Text>
					<Text style={styles.teamText}>{Translation[lang].telecom}</Text>
				</View>
			</View>
		</View>
	</ScrollView>	
	
	);
}
onBackAndroid = () => {
		const { navigator } = this.props;
		if (navigator && navigator.getCurrentRoutes().length > 1) {
			BackAndroid.removeEventListener('Back', this.onBackAndroid);
			navigator.pop();
			return true;
		} else {
			return false;
		}
  };
}

const styles = StyleSheet.create({
   container: {
      marginTop:height*0.08,
      flex:1,
   },
   gradient:{
      width: width,
      flexDirection:'row',
      paddingTop:15,
      paddingBottom:15
   },
   header: {
      alignItems: 'center',
      marginBottom:15,
   },
   headerIcon: {
      width:width/4,
      alignItems:'center',
      alignSelf:'center',
   },
   logoLeft:{
      transform: [{ rotate: '-25deg'}],
      width: 70,
      height: 70,
   },
   title: {
      width:width/2,
      fontSize: 25,
      textAlign: 'center',
      color: '#333333',
   },
   logoRight:{
      width: 70,
      height: 70,
   },
   
   containerV:{
		flexDirection: 'column',
		marginLeft:width*0.03,
		marginRight:width*0.03,
		alignItems:'center',
		justifyContent: 'center',
   },
   element1:{
	   width: width*1,
		height:width*0.15,
		alignItems:'center',
		justifyContent: 'center',
   },
   element2:{
		width: width*0.6,
		height:width*0.2,
		margin:width*0.01,
   },
   elementT:{
		width: width*1,
		height:width*0.37,
		alignItems:'center',
		justifyContent: 'center',
   },
  pictureI:{
    width:width*0.6,
    height:width*0.13,
  },
  pictureE:{
    width:width*0.48,
    height:width*0.17,
  },
  pictureC:{
    width:width*0.45,
    height:width*0.15,
  },
  pictureT:{
    width:width*0.3,
    height:width*0.37,
  },
  name:{
	  textAlign:'center',
		fontWeight: "bold",	  
  },
  teamText:{
	textAlign:'center',  
  },

});