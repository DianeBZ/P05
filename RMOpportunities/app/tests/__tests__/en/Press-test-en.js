// __tests__/Press-test-en.js
import 'react-native';
import React from 'react';

global.lang = 'en';
import Press from '../../../main/Press';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(
    <Press />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
