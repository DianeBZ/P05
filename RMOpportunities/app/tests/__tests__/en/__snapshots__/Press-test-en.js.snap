// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F2F2F2",
      "flex": 1,
      "marginTop": 106.72,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "marginBottom": 15,
      }
    }
  >
    <BVLinearGradient
      colors={
        Array [
          4279783799,
          4294365330,
        ]
      }
      end={
        Object {
          "x": 1,
          "y": 1,
        }
      }
      locations={
        Array [
          0,
          1,
        ]
      }
      start={
        Object {
          "x": 0,
          "y": 1,
        }
      }
      style={
        Object {
          "flexDirection": "row",
          "paddingBottom": 15,
          "paddingTop": 15,
          "width": 750,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "width": 187.5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "transform": Array [
                Object {
                  "rotate": "-25deg",
                },
              ],
              "width": 70,
            }
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#333333",
            "fontSize": 25,
            "textAlign": "center",
            "width": 375,
          }
        }
      >
        What the press says
      </Text>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "width": 187.5,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 70,
              "width": 70,
            }
          }
        />
      </View>
    </BVLinearGradient>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 22.5,
            "marginRight": 22.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 112.5,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 97.5,
                  "width": 450,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 150,
              "margin": 7.5,
              "width": 450,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Info chimie magazine 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 22.5,
            "marginRight": 22.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 112.5,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 127.50000000000001,
                  "width": 360,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 150,
              "margin": 7.5,
              "width": 450,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Eclaira  
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            RMOpportunities winner of the first edition of LaFabrique contest by Aviva
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 22.5,
            "marginRight": 22.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 112.5,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 112.5,
                  "width": 337.5,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 150,
              "margin": 7.5,
              "width": 450,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            CCI Paris Ile de France 
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            Chemical innovation and sustainable development driven by RMOpportunities
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "column",
            "justifyContent": "center",
            "marginLeft": 22.5,
            "marginRight": 22.5,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "height": 277.5,
              "justifyContent": "center",
              "width": 750,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                },
                undefined,
              ]
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Image
              source={1}
              style={
                Object {
                  "height": 277.5,
                  "width": 225,
                }
              }
            />
          </View>
        </View>
        <View
          style={
            Object {
              "height": 150,
              "margin": 7.5,
              "width": 450,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "fontWeight": "bold",
                "textAlign": "center",
              }
            }
          >
            Télécom Saint-Etienne  
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "textAlign": "center",
              }
            }
          >
            RMOpportunities wins the grand prize of the startup contest AVIVA
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
